default.vcl.tmpl

```diff
@@ -205,6 +205,11 @@ sub vcl_backend_response {
   if (beresp.http.Surrogate-Control ~ "ESI/1.0") {
     set beresp.do_esi = true;
     set beresp.do_gzip = true;
+    
+    # Disable ETag for ESI pages as the ETag of subrequests doesn't propagate to the parent
+    # Pages served would always rely on the cache of the main page and therefore
+    # never refresh, even if the subrequest is updated
+    unset beresp.http.ETag;
```

Comment:
Instead of doing that in `vcl_backend_response`, we should do that within `vcl_deliver`.
It's always the main page which declares `ESI` support and therefore we can move the condition in there. 

```
if (resp.http.Surrogate-Control ~ "ESI/1.0") {
  unset resp.http.ETag;
  unset resp.http.Last-Modified;
  unset resp.http.Surrogate-Control;
}
```